src/api/store/community.py

```diff
@@ -179,31 +217,56 @@ def _update_real_estate_units_with_community(self, community):
     Find any real estate units in the database which are located in this community,
     and update the link to the community.
     """ 
+    userProfiles = UserProfile.objects.prefetch_related('real_estate_units').filter(is_deleted=False)
     reus = RealEstateUnit.objects.all().select_related('address')
     print("Updating "+str(reus.count())+" RealEstateUnits")
     
-    for reu in reus:
-      if reu.address:
-        zip = reu.address.zipcode
-        if not isinstance(zip,str) or len(zip)!=5:
-          print("REU invalid zipcode: address = "+str(reu.address))
-          zip = "00000"
-          reu.address.zipcode = zip
-        
-        if is_reu_in_community(reu, community):
-          print("Adding the REU with zipcode " + zip + " to the community " + community.name)
-          reu.community = community
-          reu.save()
-
-        elif reu.community and reu.community.id == community.id:
-          # this could be the case if the community was made smaller
-          print("REU not located in this community, but was labeled as belonging to the community")
-          reu.community = None
-          reu.save()
-
-      else:
-        print("RealEstateUnit without address - do backfill first")
-        print(reu.location)
+    # loop over profiles and realEstateUnits associated with them
+    for userProfile in userProfiles:
+
+      for reu in userProfile.real_estate_units.all():
+        if reu.address:
+          zip = reu.address.zipcode
+          if not isinstance(zip,str) or len(zip)!=5:
+            address_string = str(reu.address)
+            print("REU invalid zipcode: address = "+address_string)
+
+            # temporary fix
+            if address_string.find("carlisle")>=0:
```

Comment:
please let's not be doing hardcoded stuff here please.  Even if you need this in the mean time, i would put this in a dictionary file so you have a faster look up than chaining so many ifs.  Also put it in the utils.py

Reply:
@Opoku-Agyemang   Agreed.  I moved all those cases into ZIPCODE_FIXES.json which won't get committed to github. Will slack it to you.